File: 2404-most-frequent-even-element/2404-most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int,int>mp;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i]% 2 == 0){
            mp[nums[i]]++;
            }
        }
        int maxfreq = 0;
        int ans = -1;
        for(auto it : mp){
            int a = it.first;
            int b = it.second;

            if(b > maxfreq){
                maxfreq = b;
                ans = a;
            }
            else if(b == maxfreq && a < ans){
                ans = a;
            }
        }
        return ans;
    }
};

File: 2404-most-frequent-even-element/2404-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        int n = nums.length;
        for(int i = 0; i < n; i++){
            if(nums[i] % 2 == 0){
                mp.put(nums[i], mp.getOrDefault(nums[i], 0) + 1);
            }
        }
        int ans = -1;
        int maxfreq = 0;
        for(Map.Entry<Integer,Integer> en : mp.entrySet()){
            int num = en.getKey();
            int  freq = en.getValue();

            if(freq > maxfreq){
                maxfreq = freq;
                ans = num;
            }else if(maxfreq == freq && num < ans){
                ans = num;
            }
        }
            return ans;
    }
}